### PR TITLE
New module win_mssql_login_server_role: add or remove a server role to a sql login

### PR DESCRIPTION
**Pull request by dlazz**
##### SUMMARY
This module allows to add or remove a server role to a MSSSQL server login.

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
 win_mssql_login_server_role

##### ANSIBLE VERSION
```
2.7
```

##### ADDITIONAL INFORMATION
The best way to use this module is with win_mssql_login module (added with another pull request)

Tasks:
```
- name: create new sql login
  win_mssql_login:
    server_instance: MYSQLSERVER
    login_name: john_doe
    password: "{{ my_super_secret }}"
    state: present

- name: dd server role
  win_mssql_login_server_role:
    server_instance: MYSQLSERVER
    login_name: john_doe
    server_role: sysadmin
```
Output:
```
TASK [create new sql login] **********************************************************************************************
changed: [ansible] => {"changed": true, "output": "login john_doe created"}

TASK [add server role] **********************************************************************************************
changed: [ansible] => {"changed": true, "output": "role added", "role": "sysadmin"}
```

### CODE REVIEW

**Comment by dagwieers** [comment]
Beware that [the deadline](https://github.com/ansible/ansible/blob/devel/docs/docsite/rst/roadmap/ROADMAP_2_7.rst#release-schedule) for getting new features/modules accepted in Ansible v2.7 is nearing, it is set to either 2018-08-23 or 2018-08-30. If you are blocked, or you need feedback, please discuss on IRC channel #ansible-windows or add a comment to the [Windows Working Group meeting agenda](https://github.com/ansible/community/issues/294) to get it resolved.

File: lib/ansible/modules/windows/win_mssql_login_server_role.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    required: yes
+  state:
+    description:
+      - If C(present), adds a server role
+      - If C(absent), removes a server role
+    default: present
+author:
+  - Daniele Lazzari
+notes:
+  - If the user that runs ansible has the appropriated permission on the db,
```
Comment (jhawkesworth) [inline lib/ansible/modules/windows/win_mssql_login_server_role.py]: same comment as on #43866 
is 'appropriated' a typo? I think 'appropriate' is intended perhaps?
Also I would not use 'db' but instead use full word 'database' in documentation.

**Comment by jborean93** [comment]
I don't know how I feel about this module. In reality what I think should happen is

* Create a module that manage's SQL roles and potentially add support for controlling the membership in this module
* Add support to your `win_mssql_login` module to specify a role(s) that the login is a member of

What are your thoughts on this?

**Comment by dagwieers** [comment]
During the [next Windows WG meeting](https://github.com/ansible/community/issues/420) on 2019-01-29 we will review and decide on the inclusion of this module in Ansible v2.8. Your presence in the meeting could be helpful.

*If there's anything you wanted to improve/fix, please do so before the meeting.*

**Comment by nitzmahone** [comment]
After some discussion at the Windows Working Group module review, we're a little nervous to take on a set of SQL Server modules into the core distribution right now, since we can't sanely test against a SQL Server  in CI today. This would be a good candidate for the new collections feature that's coming in Ansible 2.8, which would allow these modules to be hosted on galaxy.ansible.com and easily installed/upated by users on their Ansible controller.

**Comment by dlazz** [comment]
Ok, thank you all. I'll try the new feature.

**Pull request closed without merging**